### DISCUSSION TITLE
[7일차 -윤경]props와 state의 차이

### DISCUSSION THREAD

**Discussion by skyhanull** [zzangCS/zzangInterview#26] [프론트 면접 질문]
props와 state의 차이는?

**Comment by skyhanull** [comment]
props는 부모컴포넌트에서 자식컴포넌트로 데이터를 전달하는 것을 말하며 읽기전용입니다

state는 컴포넌트안에서 데이터를 변경할 수 있도록 하는 것입니다

**Comment by ez1n** [comment]
- **props**

> 부모 컴포넌트가 자식 컴포넌트에 전달하는 데이터

⇒ 읽기 전용이기 때문에 컴포넌트 내부에서 props를 수정해서는 안된다.
⇒ props를 받는 함수형 컴포넌트에 인자를 정의하면 props를 속성으로 가지는 객체 object가 해당 인자로 전달된다.

- **state**

> 컴포넌트 내부의 동적 데이터

⇒ 여러개의 자식으로부터 데이터를 모으거나 두 개의 자식 컴포넌트들이 서로 통신하기 위해서 부모 컴포넌트에 state를 정의한다.

**Reply by ez1n** [reply]
https://velog.io/@soyi47/React-Component-props-state

**Comment by smarfy99** [comment]
props는 readonly로 부모에서 자식으로 단방향 데이터이다.
반면 state는 값을 언제든 변경할 수 있고, 값이 변할 때마다 재렌더링된다
